drivers/packet/packet.go

```diff
@@ -0,0 +1,392 @@
+	}
+
+	return nil
+}
+
+func (d *Driver) AuthorizePort(ports []*drivers.Port) error {
+	return nil
+}
+
+func (d *Driver) DeauthorizePort(ports []*drivers.Port) error {
+	return nil
+}
+
+func (d *Driver) GetMachineName() string {
+	return d.MachineName
+}
+
+func (d *Driver) GetSSHHostname() (string, error) {
+	return d.GetIP()
+}
+
+func (d *Driver) GetSSHKeyPath() string {
+	return filepath.Join(d.StorePath, "id_rsa")
+}
+
+func (d *Driver) GetSSHPort() (int, error) {
+	if d.SSHPort == 0 {
+		d.SSHPort = 22
+	}
+
+	return d.SSHPort, nil
+}
+
+func (d *Driver) GetSSHUsername() string {
+	if d.SSHUser == "" {
+		d.SSHUser = "root"
+	}
+
+	return d.SSHUser
+}
+
+func (d *Driver) PreCreateCheck() error {
+	if !stringInSlice(d.OperatingSystem, getOsFlavors()) {
+		return fmt.Errorf("specified --packet-os not one of %v", strings.Join(getOsFlavors(), ", "))
+	}
+
+	client := d.getClient()
+	facilities, _, err := client.Facilities.List()
+	if err != nil {
+		return err
+	}
+	for _, facility := range facilities {
+		if facility.Code == d.Facility {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("packet requires a valid facility")
+}
+
+func (d *Driver) Create() error {
+	log.Infof("Creating SSH key...")
+
+	key, err := d.createSSHKey()
+	if err != nil {
+		return err
+	}
+
+	d.SSHKeyID = key.ID
+
+	client := d.getClient()
+	createRequest := &packngo.DeviceCreateRequest{
+		HostName:     d.MachineName,
+		Plan:         d.Plan,
+		Facility:     d.Facility,
+		OS:           d.OperatingSystem,
+		BillingCycle: d.BillingCycle,
+		ProjectID:    d.ProjectID,
+		UserData:     d.UserData,
+		Tags:         d.Tags,
+	}
+
+	log.Infof("Provisioning Packet server...")
+	newDevice, _, err := client.Devices.Create(createRequest)
+	if err != nil {
+		return err
+	}
+	t0 := time.Now()
+
+	d.DeviceID = newDevice.ID
+
+	for {
+		newDevice, _, err = client.Devices.Get(d.DeviceID)
+		if err != nil {
+			return err
+		}
+
+		for _, ip := range newDevice.Network {
+			if ip.Public && ip.Family == 4 {
+				d.IPAddress = ip.Address
+			}
+		}
+
+		if d.IPAddress != "" {
+			break
+		}
+
+		time.Sleep(1 * time.Second)
+	}
+
+	log.Infof("Created device ID %s, IP address %s",
+		newDevice.ID,
+		d.IPAddress)
+
+	log.Infof("Waiting for Provisioning...")
+	stage := float32(0)
+	for {
+		newDevice, _, err = client.Devices.Get(d.DeviceID)
+		if err != nil {
+			return err
+		}
+		if newDevice.State == "provisioning" && stage != newDevice.ProvisionPer {
+			stage = newDevice.ProvisionPer
+			log.Infof("Provisioning %v%% complete", newDevice.ProvisionPer)
+		}
+		if newDevice.State == "active" {
+			log.Infof("Device State: %s", newDevice.State)
+			break
+		}
+		time.Sleep(10 * time.Second)
+	}
+
+	log.Infof("Provision time: %v.\n", time.Since(t0))
+
+	log.Debug("Waiting for SSH...")
+	if err := ssh.WaitForTCP(fmt.Sprintf("%s:%d", d.IPAddress, 22)); err != nil {
+		return err
+	}
+
+	switch d.OperatingSystem {
```

Comment:
Why is this updating?  I'm not sure we want to force updates on users.  IMO this should be handled either in the base image or as a post-install step from the user (i.e. `docker-machine ssh machine -- sudo apt-get -y upgrade`).